libsolidity/analysis/TypeChecker.cpp

```diff
@@ -3816,10 +3822,10 @@ void TypeChecker::endVisit(UsingForDirective const& _usingFor)
 				)
 			);
 
-		FunctionType const* functionType = dynamic_cast<FunctionType const&>(*functionDefinition.type()).withBoundFirstArgument();
-		solAssert(functionType && functionType->selfType(), "");
+		FunctionType const* functionTypeAsBound = functionType->withBoundFirstArgument();
+		solAssert(functionTypeAsBound && functionTypeAsBound->selfType(), "");
```

Comment:
```suggestion
		FunctionType const* functionTypeWithBoundFirstArgument = functionType->withBoundFirstArgument();
		solAssert(functionTypeWithBoundFirstArgument && functionTypeWithBoundFirstArgument->selfType(), "");
```
If you look at https://github.com/ethereum/solidity/pull/13741, it seems as though we generally avoid the `asBound` nomenclature for functions where the first argument is bound, although @cameel will have to be the judge of that. I've also seen `FunctionType const* withBoundFirstArgument` in other files, so it seems as though our variable naming is not super consistent.

Reply:
I think your suggestion is better and more consistent.